kamu-core/src/domain/repos/local_dataset_repository.rs

```diff
@@ -51,6 +52,12 @@ pub trait LocalDatasetRepository: DatasetRegistry + Sync + Send {
         dataset_ref: &DatasetRefLocal,
     ) -> Result<Arc<dyn Dataset>, GetDatasetError>;
 
+    async fn get_dataset_with_external_url(
```

Comment:
As discussed on Slack - my suggestion is to remove this in favor of having existing `get_dataset()` method return datasets with external URLs pre-configured, which can be achieved by having a S3-specific implementation of the `LocalDatasetRepository`.

Reply:
Got rid of this method